### PR TITLE
Docs: Updated inline and SDK docs

### PR DESCRIPTION
**Pull request by elitan**

### CODE REVIEW

**Review by plmercereau — APPROVED** [review]

File: packages/react/src/hooks/common.ts [hunk 1]
```diff
@@ -154,6 +176,29 @@ const Component = () => {
 };
 ```
 */
+/**
```
Comment (plmercereau) [inline packages/react/src/hooks/common.ts:179]: `useSignOut` tsdoc is declared twice

File: packages/react/src/hooks/common.ts [hunk 2]
```diff
@@ -154,6 +176,29 @@ const Component = () => {
...
+ * Use the hook `useSignOut` to sign out the user.
+ *
+ * @example
+ * ```tsx
+ * import { useSignOut, useAuthenticated } from '@nhost/react'
+ *
+ * const Component = () => {
+ *   const { signOut } = useSignOut()
+ *   const isAutenticated = useAuthenticated()
```
Comment (plmercereau) [inline packages/react/src/hooks/common.ts:188]: `isAutenticated` -> `isAuthenticated`

File: packages/react/src/hooks/registration.ts [hunk 3]
```diff
@@ -40,64 +40,22 @@ interface SignUpEmailPasswordHook {
 }
 
 /**
- * Email and Password Sign-Up
+ * Use the hook `useSIgnUpEmailPassword` to sign up a user using email and password.
```
Comment (plmercereau) [inline packages/react/src/hooks/registration.ts:43]: `useSIgnUpEmailPassword` -> `useSignUpEmailPassword`

**Review by plmercereau — CHANGES_REQUESTED** [review]

File: docs/docs/reference/docgen/javascript/storage/content/hasura-storage-client/content/01-upload.mdx [hunk 4]
```diff
@@ -3,16 +3,16 @@
 title: upload()
 sidebar_label: upload()
 slug: /reference/javascript/storage/upload
-description: Use `.upload` to upload a file.
-custom_edit_url: https://github.com/nhost/nhost/edit/main/packages/hasura-storage-client.ts#L32
+description: Use `nhost.storage.upload` to upload a file. The `file` must be of type `File` (https://developer.mozilla.org/en-US/docs/Web/API/File).
+custom_edit_url: https://github.com/nhost/nhost/edit/main/packages/hasura-storage-client.ts#L41
```
Comment (szilarddoro) [inline docs/docs/reference/docgen/javascript/storage/content/hasura-storage-client/content/01-upload.mdx:7]: Something is not right with storage edit URLs. They are not pointing to `packages/hasura-storage-js/src/hasura-storage-client.ts` file.

File: docs/docs/reference/docgen/javascript/storage/content/hasura-storage-client/content/01-upload.mdx [hunk 5]
```diff
@@ -3,16 +3,16 @@
...
 ---
 
 # `upload()`
 
-Use `.upload` to upload a file.
+Use `nhost.storage.upload` to upload a file. The `file` must be of type `File` (https://developer.mozilla.org/en-US/docs/Web/API/File).
```
Comment (szilarddoro) [inline docs/docs/reference/docgen/javascript/storage/content/hasura-storage-client/content/01-upload.mdx:12]:
```markdown
[`File`](https://developer.mozilla.org/en-US/docs/Web/API/File)
```

File: packages/react/src/hooks/authentication.ts [hunk 6]
```diff
@@ -459,7 +395,7 @@ export const useSignInAnonymous = (): SignInAnonymousHookResult => {
 }
 
 /**
- * Hook that returns OAuth provider URLs
+ *  Use the hook `useProviderLink` to get a OAuth provider URL that can be used to sign in users.
```
Comment (szilarddoro) [inline packages/react/src/hooks/authentication.ts:398]:
```diff
- a OAuth
+ an OAuth
```

File: packages/react/src/hooks/user.ts [hunk 7]
```diff
@@ -549,37 +495,25 @@ interface SendVerificationEmailHook {
 }
 
 /**
- * Send email verification
+ * Use the hook `useSendVerificationEmail` to send a verification email. The verification email is sent to the user's email address and inclides a link to verify the email address.
```
Comment (szilarddoro) [inline packages/react/src/hooks/user.ts:498]:
```diff
- inclides
+ includes
```

File: packages/hasura-auth-js/src/hasura-auth-client.ts [hunk 8]
```diff
@@ -78,17 +78,17 @@ export class HasuraAuthClient {
...
-   * Use `signUp` to sign up users using email an password.
-   *
-   * If you want to sign up a user using magic link or a social provider, use
-   * the `signIn` function instead.
+   * Use `nhost.auth.signUp` to sign up a user using email and password. If you want to sign up a user using passwordless email (Magic Link), SMS, or an OAuth provider, use the `signIn` function instead.
    *
    * @example
    * ```ts
-   * auth.signUp({email, password}); // email password
+   * nhost.auth.signIn({
```
Comment (szilarddoro) [inline packages/hasura-auth-js/src/hasura-auth-client.ts:85]: `nhost.auth.signUp`

**Review by szilarddoro — APPROVED** [review]
🦾

**Pull request merged**